src/controllers/time.rs

```diff
@@ -78,10 +78,8 @@ impl TimeController {
 
         // Check if we have the "TimeSlow" powerup
         let mut time_slow = false;
-        if let Some(powerup) = state.world.player.powerup {
-            if powerup == PowerupKind::TimeSlow {
-                time_slow = true;
-            }
+        if let Some(PowerupKind::TimeSlow) = state.world.player.powerup {
+            time_slow = true;
         }
```

Comment:
What about `let time_slow = state.world.player.powerup == Some(PowerupKind::TimeSlow)`?

Reply:
Ha! It seems I continue to miss the obvious... 😅